Shared/Samples/Realistic lighting and shadows/RealisticLightingAndShadowsView.swift

```diff
@@ -0,0 +1,141 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct RealisticLightingAndShadowsView: View {
+    /// The view model for this sample.
+    @StateObject private var model = Model()
+    
+    /// A Boolean value indicating whether the settings view should be presented.
+    @State var isShowingSettings = false
+    
+    /// The date second value controlled by the slider.
+    @State private var dateSecond: Float = Model.dateSecondsAfternoon
+    
+    @State private var dateTimeText: String = DateFormatter.localizedString(
+        from: Calendar.current.startOfDay(for: Date()),
+        dateStyle: .medium,
+        timeStyle: .short
+    )
+    
+    /// The atmosphere effect of the scene view.
+    @State private var atmosphereEffect: SceneView.AtmosphereEffect = .realistic
+    
+    /// The sun lighting mode of the scene view.
+    @State private var lightingMode: SceneView.SunLighting = .lightAndShadows
+    
+    /// The sun date that gets passed into the scene view.
+    @State private var sunDate: Date = Calendar.current.startOfDay(for: Date()).advanced(by: TimeInterval(Model.dateSecondsAfternoon))
+    
+    var body: some View {
+        VStack {
+            SceneView(scene: model.scene)
+                .atmosphereEffect(atmosphereEffect)
+                .sunLighting(lightingMode)
+                .sunDate(sunDate)
+            
+            VStack {
+                Slider(value: $dateSecond, in: Model.dateSecondValueRange) {
+                } minimumValueLabel: {
+                    Text("AM")
+                } maximumValueLabel: {
+                    Text("PM")
+                }
+                .frame(maxWidth: 540)
+                .onChange(of: dateSecond, perform: sliderValueChanged(toValue:))
+            }
+            .padding()
+        }
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                Text(dateTimeText)
+                Button("Mode") {
+                    isShowingSettings = true
+                }
+                .alert("Choose a lighting mode for the scene view.", isPresented: $isShowingSettings) {
+                    ForEach(SceneView.SunLighting?.allCases, id: \.self) { mode in
+                        Button(mode.label) {
+                            lightingMode = mode!
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Handle slider value changed event and set the scene view's sun date.
```

Comment:
```suggestion
    /// Handles slider value changed event and set the scene view's sun date.
```
btw We gonna discuss the verb tense this Wednesday.